crates/astria-conductor/src/celestia/mod.rs

```diff
@@ -170,6 +175,16 @@ impl Reader {
     async fn initialize(
```

Comment:
This `initialize` function also fetches the sequencer chain ID a few lines down: after all, the celestia reader does two things:

1. read blobs from celestia
2. verify the sequencer block information extracted from the blobs against the connected sequencer.

This function only checks that the celestia chain ID matches, but not if the sequencer chain ID against the blocks will be verified matches!

Please update this method to:

1. also check the sequencer chain ID,
2. and to perform all the 3 checks (test for celestia chain ID, waiting for the executor, test for sequencer chain ID) concurrently using `try_join!`